sdk/program/src/loader_v4.rs

```diff
@@ -41,9 +46,147 @@ impl LoaderV4State {
     }
 }
 
+pub fn is_write_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 0 == instruction_data[0]
+}
+
+pub fn is_truncate_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 1 == instruction_data[0]
+}
+
+pub fn is_deploy_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 2 == instruction_data[0]
+}
+
+pub fn is_retract_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 3 == instruction_data[0]
+}
+
+pub fn is_transfer_authority_instruction(instruction_data: &[u8]) -> bool {
+    !instruction_data.is_empty() && 4 == instruction_data[0]
+}
+
+/// Returns the instructions required to initialize a program/buffer account.
+pub fn create_buffer(
+    payer_address: &Pubkey,
+    buffer_address: &Pubkey,
+    lamports: u64,
+    authority: &Pubkey,
+    new_size: u32,
+    recipient_address: &Pubkey,
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account(payer_address, buffer_address, lamports, 0, &id()),
+        Instruction::new_with_bincode(
+            id(),
+            &LoaderV4Instruction::Truncate { new_size },
+            vec![
+                AccountMeta::new(*buffer_address, true),
+                AccountMeta::new_readonly(*authority, true),
+                AccountMeta::new(*recipient_address, false),
```

Comment:
`recipient_address` is not needed and should be `payer_address` here as well.

Reply:
I thought about it. The caller can pass the payer address as the `recipient_address`.